pkg/controller/controller_test.go

```diff
@@ -0,0 +1,167 @@
+			PV:            pv(2),
+			CreateObjects: true,
+		},
+		{
+			Name:          "Resize PVC, no FS resize",
+			PVC:           pvc(2, 1),
+			PV:            pv(1),
+			CreateObjects: true,
+		},
+		{
+			Name:          "Resize PVC with FS resize",
+			PVC:           pvc(2, 1),
+			PV:            pv(1),
+			CreateObjects: true,
+			NodeResize:    true,
+		},
+	} {
+		client := csi.NewMockClient(test.NodeResize, true, true)
+
+		initialObjects := []runtime.Object{}
+		if test.CreateObjects {
+			if test.PVC != nil {
+				initialObjects = append(initialObjects, test.PVC)
+			}
+			if test.PV != nil {
+				initialObjects = append(initialObjects, test.PV)
+			}
+		}
+
+		kubeClient, informerFactory := fakeK8s(initialObjects)
+		pvInformer := informerFactory.Core().V1().PersistentVolumes()
+		pvcInformer := informerFactory.Core().V1().PersistentVolumeClaims()
+
+		for _, obj := range initialObjects {
+			switch obj.(type) {
+			case *v1.PersistentVolume:
+				pvInformer.Informer().GetStore().Add(obj)
+			case *v1.PersistentVolumeClaim:
+				pvcInformer.Informer().GetStore().Add(obj)
+			default:
+				t.Fatalf("Test %s: Unknown initalObject type: %+v", test.Name, obj)
+			}
+		}
+
+		csiResizer, err := resizer.NewResizerFromClient(client, 15*time.Second, kubeClient, informerFactory)
+		if err != nil {
+			t.Fatalf("Test %s: Unable to create resizer: %v", test.Name, err)
+		}
+
+		driverName, _ := client.GetDriverName(context.TODO())
+		controller := NewResizeController(driverName, csiResizer, kubeClient, time.Second, informerFactory)
+		err = controller.(*resizeController).syncPVC(fmt.Sprintf("%s/%s", test.PVC.Namespace, test.PVC.Name))
+		if err != nil {
+			t.Fatalf("Test %s: Unexpected error: %v", test.Name, err)
+		}
+	}
+}
+
+func invalidPVC() *v1.PersistentVolumeClaim {
+	pvc := pvc(1, 1)
+	pvc.ObjectMeta.Name = ""
+	pvc.ObjectMeta.Namespace = ""
+
+	return pvc
+}
+
+func quantityGB(i int) resource.Quantity {
+	q := resource.NewQuantity(int64(i*1024*1024), resource.BinarySI)
+	return *q
+}
+
+func pvc(requestGB, capacityGB int) *v1.PersistentVolumeClaim {
+	request := quantityGB(requestGB)
+	capacity := quantityGB(capacityGB)
+
+	return &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testPVC",
+			Namespace: "test",
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			Resources: v1.ResourceRequirements{
+				Requests: map[v1.ResourceName]resource.Quantity{
+					v1.ResourceStorage: request,
+				},
+			},
+			VolumeName: "testPV",
+		},
+		Status: v1.PersistentVolumeClaimStatus{
+			Phase: v1.ClaimBound,
+			Capacity: map[v1.ResourceName]resource.Quantity{
+				v1.ResourceStorage: capacity,
+			},
+		},
+	}
+}
+
+func pv(capacityGB int) *v1.PersistentVolume {
+	capacity := quantityGB(capacityGB)
+
+	return &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testPV",
+		},
+		Spec: v1.PersistentVolumeSpec{
+			Capacity: map[v1.ResourceName]resource.Quantity{
+				v1.ResourceStorage: capacity,
+			},
+		},
```

Comment:
Can we get this PV to have `CSIPersistentVolumeSource` field, so as it is supported by the resizer? As it is - mock driver's `Expand` call is never made because PVs don't have CSI field.

Reply:
Done.